cogeo_mosaic/backends/base.py

```diff
@@ -38,21 +52,34 @@ class BaseBackend(BaseReader):
     # works with WebMercator (mercantile) for now.
     tms: TileMatrixSet = attr.ib(init=False, default=WEB_MERCATOR_TMS)
 
+    bounds: Tuple[float, float, float, float] = attr.ib(
+        init=False, default=(-180, -90, 180, 90)
+    )
+    minzoom: int = attr.ib(init=False, default=0)
+    maxzoom: int = attr.ib(init=False, default=30)
+
+    mosaic_def: MosaicJSON = attr.ib(init=False)
+
     _backend_name: str
     _file_byte_size: Optional[int] = 0
 
-    @mosaic_def.validator
-    def _check_mosaic_def(self, attribute, value):
-        if value is not None:
-            self.mosaic_def = MosaicJSON(**dict(value))
+    @mode.validator
+    def _check_mode(self, attribute, value):
+        if value not in ["r", "r+", "w"]:
+            raise ValueError(f"Invalid {value} mode. MUST be one of (r, r+, w).")
 
     def __attrs_post_init__(self):
         """Post Init: if not passed in init, try to read from self.path."""
-        self.mosaic_def = self.mosaic_def or self._read(**self.backend_options)
+        # here we can warns if the mosaic already exists and is openned in `w` mode.
+        # if self.mode == "w" and self.mosaic_exits:
+        #     raise MosaicExistsError("Mosaic already exist, use `overwrite=True`.")
 
-        self.minzoom = self.mosaic_def.minzoom
-        self.maxzoom = self.mosaic_def.maxzoom
-        self.bounds = self.mosaic_def.bounds
+        if self.mode in ["r", "r+"]:
```

Comment:
I think you should just do `if 'r' in self.mode`. For example, isn't `rw` a valid mode?

Reply:
I think I went the way it is in rasterio
`r`: read-only
`r+`: read+update
`w`: write

🤷‍♂️